simnode/generator.py

```diff
@@ -85,50 +87,166 @@ def _calculate_parallel_coords(offset: float, line_width: float) \
+                        coords = lstr.coords.xy
+
+                        # for plotting the generated road, useful for debugging
+                        '''import matplotlib.pyplot as plt
+                        plt.plot(offset_sub_lines[0], offset_sub_lines[1])
+                        plt.ylabel("first offset: " + str(offset[0]))
+                        plt.show()
+                        '''
+                    # NOTE The parallel LineString may have a different number of points than initially given
+                    num_coords = len(coords[0])
+                    z_vals = repeat(0.01, num_coords)
+                    marking_widths = repeat(line_width, num_coords)
+                    return list(zip(coords[0], coords[1], z_vals, marking_widths))
+
                 # Draw side lines
                 num_lines = road.left_lanes + road.right_lanes + 1
                 initial_line_offsets = [d - (road_width / 2) for d in linspace(0, road_width, num=num_lines)]
                 side_line_offset = 1.5 * self.add_roads_to_scenario.line_width
                 left_side_line = Road('line_white', rid=road.rid + "_left_line")
-                left_side_line_nodes = _calculate_parallel_coords(
-                    initial_line_offsets[0] + side_line_offset, self.add_roads_to_scenario.line_width)
+                offset_list_line_left = _calculate_offset_list(relative_offset=-0.5,
+                                                                absolute_offset=side_line_offset)
+                left_side_line_nodes = _calculate_parallel_coords_list(offset_list_line_left,
+                                                                        self.add_roads_to_scenario.line_width)
                 if left_side_line_nodes:
                     left_side_line.nodes.extend(left_side_line_nodes)
                     scenario.add_road(left_side_line)
                 else:
                     _logger.warning("Could not create left side line")
                 right_side_line = Road('line_white', rid=road.rid + "_right_line")
-                right_side_line_nodes = _calculate_parallel_coords(
-                    initial_line_offsets[-1] - side_line_offset, self.add_roads_to_scenario.line_width)
+                offset_list_line_right = _calculate_offset_list(relative_offset=.5, absolute_offset=-side_line_offset)
+                right_side_line_nodes = _calculate_parallel_coords_list(offset_list_line_right,
+                                                                        self.add_roads_to_scenario.line_width)
                 if right_side_line_nodes:
                     right_side_line.nodes.extend(right_side_line_nodes)
                     scenario.add_road(right_side_line)
                 else:
                     _logger.warning("Could not create right side line")
 
                 # Draw line separating left from right lanes
+                # FIXME rightLanes cannot be 0, is this by design?
```

Comment:
The fact that rightLanes can not be 0 results only from the definition in the XSD. Since this piece of code and the XSD are quite far away from each other and many things may happen until the data reaches this point, this if improves the safety. Besides it explicitly states conditions which have to be fulfilled in order to draw the lines.

Reply:
So yes. This check is wanted.